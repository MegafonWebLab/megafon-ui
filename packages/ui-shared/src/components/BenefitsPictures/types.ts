export interface IBenefit {
    /** Заголовок бенефита */
    title: string;
    /** Основной текст в бенефите */
    text: string;
    /** Изображение */
    img: string;
}

type TGridSizeValues = '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9' | '10' | '11' | '12';

export type GridConfig = {
    wide?: TGridSizeValues;
    desktop?: TGridSizeValues;
    tablet?: TGridSizeValues;
    leftOffsetWide?: TGridSizeValues;
    leftOffsetDesktop?: TGridSizeValues;
    leftOffsetTablet?: TGridSizeValues;
    rightOffsetDesktop?: TGridSizeValues;
    rightOffsetTablet?: TGridSizeValues;
    rightOffsetWide?: TGridSizeValues;
};

export type GridGutterSize = 'large' | 'medium';
