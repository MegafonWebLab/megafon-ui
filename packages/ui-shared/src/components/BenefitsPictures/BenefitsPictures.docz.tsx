import img from '../../../demo/bg_light_grey.png';

const getItems = (i, image) => Array.from({ length: i }, () => ({
    title: 'Lorem ipsum',
    text: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit',
    img: image,
}));

export const twoItems = getItems(2, img);
export const threeItems = getItems(3, img);
export const fourItems = getItems(4, img);
