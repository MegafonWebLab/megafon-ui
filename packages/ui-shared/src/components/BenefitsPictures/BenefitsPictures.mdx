---
name: BenefitsPictures
menu: Constructor
route: components/benefits-pictures
---

import { Playground, Props } from 'docz';
import BenefitsPictures from './BenefitsPictures';
import { twoItems, threeItems, fourItems } from './BenefitsPictures.docz.tsx';

# BenefitsPictures

## Properties
<Props of={BenefitsPictures} />

## Included props
``` javascript
interface IBenefit {
    title: string;
    text: string;
    img: string;
}

type GridGutterSize = 'large' | 'medium';

const items = [
    {
        title: 'Lorem ipsum',
        text: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit',
        img: image.png,
    },
    {
        title: 'Lorem ipsum',
        text: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit',
        img: image.png,
    },
]
```

## Basic usage

### Left align, large grid gap
<Playground>
    <BenefitsPictures
        items={twoItems}
    />
</Playground>

<Playground>
    <BenefitsPictures
        items={threeItems}
    />
</Playground>

<Playground>
    <BenefitsPictures
        items={fourItems}
    />
</Playground>

### Left align, medium grid gap

<Playground>
    <BenefitsPictures
        gridGap="medium"
        items={twoItems}
    />
</Playground>

<Playground>
    <BenefitsPictures
        gridGap="medium"
        items={threeItems}
    />
</Playground>

<Playground>
    <BenefitsPictures
        gridGap="medium"
        items={fourItems}
    />
</Playground>

### Center align, large grid gap

<Playground>
    <BenefitsPictures
        hAlign="center"
        items={twoItems}
    />
</Playground>

<Playground>
    <BenefitsPictures
        hAlign="center"
        items={threeItems}
    />
</Playground>

<Playground>
    <BenefitsPictures
        hAlign="center"
        items={fourItems}
    />
</Playground>

### Center align, medium grid gap

<Playground>
    <BenefitsPictures
        gridGap="medium"
        hAlign="center"
        items={twoItems}
    />
</Playground>

<Playground>
    <BenefitsPictures
        gridGap="medium"
        hAlign="center"
        items={threeItems}
    />
</Playground>

<Playground>
    <BenefitsPictures
        gridGap="medium"
        hAlign="center"
        items={fourItems}
    />
</Playground>
