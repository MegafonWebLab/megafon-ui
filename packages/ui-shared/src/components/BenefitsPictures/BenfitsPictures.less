@import '~@megafon/ui-core/styles/base.less';

@b: e('.mfui-beta-benefits-pictures');

@{b} {
    &__img {
        max-width: 100%;
        margin-bottom: 32px;
    }

    &__img_h-align_center {
        display: block;
        margin-left: auto;
        margin-right: auto;
    }

    &__title {
        margin-bottom: 12px;
    }
}
