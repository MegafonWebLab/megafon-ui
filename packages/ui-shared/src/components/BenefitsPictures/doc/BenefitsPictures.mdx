---
name: BenefitsPictures
menu: Конструктор
route: components/benefits-pictures
---

import Example from './BenefitsPictures.example.mdx';
import Props from './BenefitsPictures.props.mdx';

# BenefitsPictures

<CopyToClipboardBox text="import { BenefitsPictures } from '@megafon/ui-shared';" />

<DoczTabs>
    <Example />
    <Props />
</DoczTabs>
