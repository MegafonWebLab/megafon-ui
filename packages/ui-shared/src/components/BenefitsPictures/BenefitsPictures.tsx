import * as React from 'react';
import { useCallback } from 'react';
import PropTypes from 'prop-types';
import throttle from 'lodash.throttle';
import './BenfitsPictures.less';
import { cnCreate, Grid, GridColumn, Header, Paragraph, breakpoints, throttleTime } from '@megafon/ui-core';
import { IBenefit, GridConfig, GridGutterSize } from './types';

const ONLY_LEFT_ALIGN_ITEMS_COUNT = 3;

const columnSize: GridConfig = {
    wide: '4',
    desktop: '4',
    tablet: '5',
};

export interface IBenefitsPicturesProps {
    /** Данные для бенефитов */
    items: IBenefit[];
    /** Горизонтальное выравнивание */
    hAlign?: 'left' | 'center';
    /** Расстояние между бенефитами */
    gridGap?: GridGutterSize;
}

const isEvenIndex = index => !((index + 1) % 2);
const getEvenOffset = index =>  isEvenIndex(index) ? '1' : undefined;
const getOddOffset = index => isEvenIndex(index) ? undefined : '1';

const getLeftConfig = (count: number, index: number): GridConfig => {
    switch (count) {
        case 2:
        case 4:
            return {
                ...columnSize,
                leftOffsetWide: getEvenOffset(index),
                leftOffsetDesktop: getEvenOffset(index),
                leftOffsetTablet: getEvenOffset(index),
            };

        default:
            return {
                ...columnSize,
                leftOffsetTablet: getEvenOffset(index),
            };
    }
};

const getCenterMediumConfig = (count: number, index: number): GridConfig => {
    switch (count) {
        case 3:
            return {
                ...columnSize,
                leftOffsetTablet: getOddOffset(index),
            };

        case 4:
            return {
                ...columnSize,
                rightOffsetWide: getOddOffset(index),
                leftOffsetWide: getEvenOffset(index),
                rightOffsetDesktop: getEvenOffset(index),
                leftOffsetDesktop: getOddOffset(index),
                rightOffsetTablet: getEvenOffset(index),
                leftOffsetTablet: getOddOffset(index),
            };

        default:
            return {
                ...columnSize,
                rightOffsetWide: getOddOffset(index),
                leftOffsetWide: getEvenOffset(index),
            };
    }
};

const getCenterLargeConfig = (count: number, index: number): GridConfig => {
    switch (count) {
        case 4:
            return {
                ...columnSize,
                leftOffsetWide: getOddOffset(index),
                rightOffsetWide: getEvenOffset(index),
                leftOffsetDesktop: getOddOffset(index),
                rightOffsetDesktop: getEvenOffset(index),
                leftOffsetTablet: getOddOffset(index),
                rightOffsetTablet: getEvenOffset(index),
            };

        case 3: {
            return {
                ...columnSize,
                leftOffsetTablet: getOddOffset(index),
            };
        }

        default:
            return {
                ...columnSize,
            };
    }
};

const getCenterConfig = (count: number, index: number, gutterSize: string): GridConfig => {
    switch (gutterSize) {
        case 'medium': {
            return getCenterMediumConfig(count, index);
        }

        default: {
            return getCenterLargeConfig(count, index);
        }
    }
};

const cn = cnCreate('mfui-beta-benefits-pictures');
const BenefitsPictures: React.FC<IBenefitsPicturesProps> = ({
    items,
    hAlign = 'left',
    gridGap = 'large',
}) => {
    const isLargeGutter = gridGap === 'large';
    const isGridCenterAlign = hAlign === 'center' && items.length !== ONLY_LEFT_ALIGN_ITEMS_COUNT;
    const [currentGutter, setCurrentGutter] = React.useState(gridGap);

    const resizeHandler = useCallback(
        () => {
            if (!isLargeGutter) {
                return;
            }

            if (window.innerWidth < breakpoints.desktopMiddleStart) {
                setCurrentGutter('medium');
            } else {
                setCurrentGutter('large');
            }
        }, []
    );

    React.useEffect(() => {
        const throttledResizeHandler = throttle(resizeHandler, throttleTime.resize);

        resizeHandler();
        window.addEventListener('resize', throttledResizeHandler);

        return () => window.removeEventListener('resize', throttledResizeHandler);
    }, []);

    return (
        <div className={cn()}>
            <Grid
                guttersLeft={currentGutter}
                hAlign={isGridCenterAlign ? 'center' : 'left'}
            >
                {items.map(({ img, title, text }, index) =>
                    <GridColumn
                        {...hAlign === 'left'
                            ? getLeftConfig(items.length, index)
                            : getCenterConfig(items.length, index, gridGap)}
                        key={index}
                    >
                        <div className={cn('item')}>
                            <img className={cn('img', { 'h-align': hAlign })}
                                 src={img}
                                 alt=""
                            />
                            <Header className={cn('title')} hAlign={hAlign} as="h3">{title}</Header>
                            <Paragraph align={hAlign}>{text}</Paragraph>
                        </div>
                    </GridColumn>
                )}
            </Grid>
        </div>
    );
};

BenefitsPictures.propTypes = {
    items: PropTypes.arrayOf(
        PropTypes.shape({
            title: PropTypes.string.isRequired,
            text: PropTypes.string.isRequired,
            img: PropTypes.string.isRequired,
        }).isRequired
    ).isRequired,
    hAlign: PropTypes.oneOf(['left', 'center']),
    gridGap: PropTypes.oneOf(['medium', 'large']),
};

export default BenefitsPictures;
