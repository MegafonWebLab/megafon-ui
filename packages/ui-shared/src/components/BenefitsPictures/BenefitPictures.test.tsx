import * as React from 'react';
import { shallow } from 'enzyme';
import BenefitsPictures, { IBenefitsPicturesProps } from './BenefitsPictures';
import { fourItems, threeItems, twoItems } from './doc/BenefitsPictures.docz';

const props: IBenefitsPicturesProps = {
    items: twoItems,
    hAlign: 'left',
    gridGap: 'large',
};

const itemsSet = [
    twoItems,
    threeItems,
    fourItems,
];

describe('<BenefitsPictures />', () => {
    it('renders BenefitsPictures', () => {
        const wrapper = shallow(<BenefitsPictures {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('render with center horizontal align', () => {
        itemsSet.forEach(items => {
            const wrapper = shallow(<BenefitsPictures {...props} items={items} hAlign={'left'} />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    it('render with medium grid gap', () => {
        itemsSet.forEach(items => {
            const wrapper = shallow(<BenefitsPictures {...props} items={items} gridGap={'medium'} />);
            expect(wrapper).toMatchSnapshot();
        });
    });
});
