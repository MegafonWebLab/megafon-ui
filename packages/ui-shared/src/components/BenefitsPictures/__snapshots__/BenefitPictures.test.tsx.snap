// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BenefitsPictures /> render with center horizontal align 1`] = `
<div
  className="mfui-beta-benefits-pictures"
>
  <Grid
    guttersLeft="large"
    hAlign="left"
    multiRow={true}
  >
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="0"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="1"
      leftOffsetDesktop="1"
      leftOffsetTablet="1"
      leftOffsetWide="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<BenefitsPictures /> render with center horizontal align 2`] = `
<div
  className="mfui-beta-benefits-pictures"
>
  <Grid
    guttersLeft="large"
    hAlign="left"
    multiRow={true}
  >
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="0"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="1"
      leftOffsetTablet="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="2"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<BenefitsPictures /> render with center horizontal align 3`] = `
<div
  className="mfui-beta-benefits-pictures"
>
  <Grid
    guttersLeft="large"
    hAlign="left"
    multiRow={true}
  >
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="0"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="1"
      leftOffsetDesktop="1"
      leftOffsetTablet="1"
      leftOffsetWide="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="2"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="3"
      leftOffsetDesktop="1"
      leftOffsetTablet="1"
      leftOffsetWide="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<BenefitsPictures /> render with medium grid gap 1`] = `
<div
  className="mfui-beta-benefits-pictures"
>
  <Grid
    guttersLeft="medium"
    hAlign="left"
    multiRow={true}
  >
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="0"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="1"
      leftOffsetDesktop="1"
      leftOffsetTablet="1"
      leftOffsetWide="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<BenefitsPictures /> render with medium grid gap 2`] = `
<div
  className="mfui-beta-benefits-pictures"
>
  <Grid
    guttersLeft="medium"
    hAlign="left"
    multiRow={true}
  >
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="0"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="1"
      leftOffsetTablet="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="2"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<BenefitsPictures /> render with medium grid gap 3`] = `
<div
  className="mfui-beta-benefits-pictures"
>
  <Grid
    guttersLeft="medium"
    hAlign="left"
    multiRow={true}
  >
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="0"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="1"
      leftOffsetDesktop="1"
      leftOffsetTablet="1"
      leftOffsetWide="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="2"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="3"
      leftOffsetDesktop="1"
      leftOffsetTablet="1"
      leftOffsetWide="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<BenefitsPictures /> renders BenefitsPictures 1`] = `
<div
  className="mfui-beta-benefits-pictures"
>
  <Grid
    guttersLeft="large"
    hAlign="left"
    multiRow={true}
  >
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="0"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
    <GridColumn
      all="12"
      desktop="4"
      flex={false}
      grow={false}
      key="1"
      leftOffsetDesktop="1"
      leftOffsetTablet="1"
      leftOffsetWide="1"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="5"
      wide="4"
    >
      <div
        className="mfui-beta-benefits-pictures__item"
      >
        <img
          alt=""
          className="mfui-beta-benefits-pictures__img mfui-beta-benefits-pictures__img_h-align_left"
          src="test-file-stub"
        />
        <Header
          as="h3"
          className="mfui-beta-benefits-pictures__title"
          color="black"
          hAlign="left"
        >
          Lorem ipsum
        </Header>
        <Paragraph
          align="left"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum, dolor sit amet consectetur adipisicing elit
        </Paragraph>
      </div>
    </GridColumn>
  </Grid>
</div>
`;
